test/jdk/java/util/jar/JarFile/LargeManifestOOMTest.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import jdk.test.lib.util.JarUtils;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.jar.JarFile;
+
+/**
+ * @test
+ * @bug 8242882
+ * @summary Verify that opening a jar file with a large manifest throws an OutOfMemoryError
+ * and not a NegativeArraySizeException
+ * @library /test/lib
+ * @run testng LargeManifestOOMTest
+ */
+public class LargeManifestOOMTest {
+
+
+    /**
+     * Creates a jar which has a large manifest file and then uses the {@link JarFile} to
+     * {@link JarFile#getManifest() load the manifest}. The call to the {@link JarFile#getManifest()}
+     * is then expected to throw a {@link OutOfMemoryError}
+     */
+    @Test
+    public void testOutOfMemoryError() throws Exception {
+        final Path jarSourceRoot = Paths.get("jar-source");
+        createLargeManifest(jarSourceRoot.resolve("META-INF"));
+        final Path jarFilePath = Paths.get("oom-test.jar");
+        JarUtils.createJarFile(jarFilePath.toAbsolutePath(), jarSourceRoot);
+        final JarFile jar = new JarFile(jarFilePath.toFile());
+        final OutOfMemoryError oome = Assert.expectThrows(OutOfMemoryError.class, () -> jar.getManifest());
+        // additionally verify that the OOM was for the right/expected reason
+        if (!"Required array size too large".equals(oome.getMessage())) {
```

Comment:
I wasn't too sure if I should add this additional check on the message of the `OutOfMemoryError`, but decided to do it anyway, given that from what I remember there were some discussions in the `core-libs-dev` list a while back on the exact messages that such OOMs should throw. So I guessed that it might be OK to rely on those messages in the tests within this project. However, I am open to removing specific check if it's considered unnecessary.

Reply:
I think it's fine either way.

Reply:
If you are going to validate the message, which I probably would not,  it would be important to make sure it document if the message is changed in JarFile::getBytes, that the test needs updated.  Otherwise it will be easy to miss.

Reply:
Hello Lance,
I decided to remove the assertion on the exception message. I have updated the PR accordingly.